src/sentry/workflow_engine/detectors/error.py

```diff
@@ -0,0 +1,173 @@
+from collections.abc import MutableMapping
+from dataclasses import dataclass
+from datetime import datetime, timedelta, timezone
+from enum import StrEnum
+from time import time
+from typing import Any, TypedDict, TypeVar
+
+from django.utils import timezone as django_timezone
+
+from sentry import analytics
+from sentry.eventstore.models import Event
+from sentry.integrations.tasks.kick_off_status_syncs import kick_off_status_syncs
+from sentry.issues import grouptype
+from sentry.metrics.middleware import MutableTags
+from sentry.models.activity import Activity
+from sentry.models.group import Group, GroupStatus
+from sentry.models.grouphistory import GroupHistoryStatus, record_group_history
+from sentry.models.groupinbox import GroupInboxRemoveAction, remove_group_from_inbox
+from sentry.models.options.project_option import ProjectOption
+from sentry.models.project import Project
+from sentry.signals import issue_resolved
+from sentry.types.activity import ActivityType
+from sentry.workflow_engine.models import DataPacket
+from sentry.workflow_engine.models.data_condition import (
+    DetectorWorkflowCondition,
+    condition_registry,
+)
+from sentry.workflow_engine.processors.detector import DetectorEvaluationResult, DetectorHandler
+from sentry.workflow_engine.types import DetectorGroupKey
+
+T = TypeVar("T")
+
+
+class ErrorDetectorConfigType(StrEnum):
+    GROUPING = "ErrorGroupingCondition"
+    RESOLVE = "ErrorResolveCondition"
```

Comment:
I feel like this is introducing a competing strategy to evaluate conditions - we should refactor so we're using DataConditions - that will allow us to simplify a lot of the logic in these classes.